File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.js
function solution(n) {
    const result = [n];  // 시작값 n을 먼저 저장
    
    while (n !== 1) {
        if (n % 2) {
            n = 3 * n + 1;
        } else {
            n = n / 2;
        }
        result.push(n);
    }
    
    return result;
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        List<Integer> collatz = new ArrayList<>();
        
        int x = n;
        collatz.add(x);
        
        while (x != 1) {
            if (x % 2 == 0) {
                x /= 2;
            } else {
                x = 3 * x + 1;
            }
            
            collatz.add(x);
        }
        
        return collatz.stream().mapToInt(Integer::intValue).toArray();
    }
}